### PR TITLE
Improve dedup using SQL query

### PR DESCRIPTION
**Pull request by gonzalesedwin1123**
Completed the ID Documents Dedup

ToDos:
- [x] Default Dedup
- [x] Phone Numbers Dedup
- [x] ID Documents Dedup

### CODE REVIEW

File: g2p_programs/models/managers/deduplication_manager.py [hunk 1]
```diff
@@ -174,21 +171,86 @@ class IDDocumentDeduplication(models.Model):
...
-                    beneficiaries
-                )
-            return len(duplicate_beneficiaries)
+                duplicate_partner_ids.append(rec["identity_partner_id"])
+
+        _logger.info("DEBUG: duplicate partner ids: %s", duplicate_partner_ids)
+
+        duplicate_beneficiaries = beneficiaries.filtered(
+            lambda rec: rec.partner_id.id in duplicate_partner_ids
+        )
```
Comment (jeremi) [inline g2p_programs/models/managers/deduplication_manager.py]: We are losing all the benefits of the optimization we did with the SQL query earlier. 
There could be millions of records in a program we cannot be filtering in odoo, it needs to be done in the DB

**Review by jeremi — CHANGES_REQUESTED** [review]

File: g2p_programs/models/managers/deduplication_manager.py (hunk 1 above)
Comment (gonzalesedwin1123) [reply g2p_programs/models/managers/deduplication_manager.py]: I have made some fixes based on your comments. Please check again.

File: g2p_programs/models/managers/deduplication_manager.py [hunk 2]
```diff
@@ -80,15 +88,15 @@ def _record_duplicate(self, manager, beneficiary_ids, reason):
                         rec.beneficiary_ids.ids,
                     )
                 )
-                # _logger.info("DEBUG! res: %s", res)
+                _logger.info("DEBUG! res: %s", res)
 
                 if 1 in res:
-                    # _logger.info("DEBUG! Update the duplicate record: %s", data)
+                    _logger.info("DEBUG! Update the duplicate record: %s", data)
```
Comment (jeremi) [inline g2p_programs/models/managers/deduplication_manager.py:94]: For debug, use logger.debug instead so it is easy to disable them.

**Review by jeremi — CHANGES_REQUESTED** [review]

**Review by jeremi — CHANGES_REQUESTED** [review]
@gonzalesedwin1123 It is good except for one view. In the program, we should list the duplicated groups like on `Programs Membership Duplicate`, not the list of program memberships that are duplicates.

**Comment by gonzalesedwin1123** [comment]
> @gonzalesedwin1123 It is good except for one view. In the program, we should list the duplicated groups like on `Programs Membership Duplicate`, not the list of program memberships that are duplicates.

I will work on this

**Pull request closed without merging**